### ISSUE TITLE
File-level expectations for files on S3 (v3 API)

### ISSUE THREAD

**Issue by SSchotten** [fivetran/great_expectations#3979]
**Is your feature request related to a problem? Please describe.**
Using the v3 API, I would like to be able to create and use file-level expectations - similar to the ones defined for the v2 API as specified [here](https://legacy.docs.greatexpectations.io/en/latest/reference/glossary_of_expectations.html#filedataasset).  

In particular, I would like to be able to create/validate expectations about whether certain filenames are amongst a daily data delivery to a few S3 (actually, Minio) staging buckets, using something like v2's `expect_file_to_exist`: [click](https://legacy.docs.greatexpectations.io/en/latest/autoapi/great_expectations/data_asset/file_data_asset/index.html#great_expectations.data_asset.file_data_asset.FileDataAsset.expect_file_to_exist). Some files are expected to *always* land in these buckets, and some of them only once in a while. If detected properly, I would want to not execute any processing steps downstream of our staging layer if any of these expected files are missing. 

**Describe the solution you'd like**
The ideal outcome would be for `expect_file_to_exist` and other file-level expectations to be evaluated the same way table/column-level expectations can now be evaluated. I have only recently started looking into using GE v3 on S3, after playing with v2 locally for a bit in 2020 - so I find it hard to formulate a more detailed solution. 

**Describe alternatives you've considered**
My current approach is to perform a daily match of all files found in the S3 staging buckets against a set of expected filenames and raise an exception on the occurrences of missing files - outside of GE.

**Comment by austiezr** [comment]
Hey @SSchotten ! Thanks for surfacing this. We're reviewing internally and will be in touch.

**Comment by SSchotten** [comment]
Just to add to the above: I've recently learned that we'll be migrating to Azure soon - so my issue still stands, but should probably be read as cloud agnostic (instead of specifically mentioning S3).

**Comment by SSchotten** [comment]
Hi @austiezr , I'm curious to hear whether the idea got dropped altogether or postponed. 

My team has worked on other topics over the past half year (so I wasn't too active in that area either), but we'll be returning to further implementing GE during the next few months. Would be very helpful to have a feature like this.

**Comment by karthiga-ge** [comment]
Hi , I'm looking for the same. I'm expecting something like library which helps to validate files in AWS , without GE deployment files. Any suggestion.

**Comment by molliemarie** [comment]
Hi @SSchotten. With the upcoming launch of Great Expectations Core (GX 1.0), we are closing old issues posted regarding previous versions. Moving forward, we will focus our resources on supporting and improving GX Core (version 1.0 and beyond). If you find that an issue you previously reported still exists in GX Core, we encourage you to resubmit it against the new version. With more resources dedicated to community support, we aim to tackle new issues swiftly. For specific details on what is GX-supported vs community-supported, you can reference our [integration and support policy](https://docs.greatexpectations.io/docs/application_integration_support/).

To get started on your transition to GX Core, check out the [GX Core  quickstart](https://docs.greatexpectations.io/docs/core/introduction/try_gx/#test-features-and-functionality) (click “Full example code” tab to see a code example). 

You can also join our upcoming community meeting on August 28th at 9am PT (noon ET / 4pm UTC) for more updates on GX Core. Go to https://greatexpectations.io/meetup and click “follow calendar” to follow the GX community calendar.

Thank you for being part of the GX community and thank you for submitting this issue. We're excited about this new chapter and look forward to your feedback on GX Core. 🤗
